File: backend/app/models/video.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Float, BigInteger, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base


class Video(Base):
    __tablename__ = "videos"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="SET NULL"), nullable=True)
    course_id = Column(Integer, ForeignKey("courses.id", ondelete="CASCADE"), nullable=True)
    order_index = Column(Integer, nullable=False, default=1)
    title = Column(String(255), nullable=True)
    filename = Column(String(255), nullable=False)
    original_filename = Column(String(255), nullable=False)
    file_path = Column(String(500), nullable=False)
    file_size = Column(BigInteger, nullable=False, default=0)
    status = Column(String(50), nullable=False, default="uploaded")
    progress = Column(Float, nullable=False, default=0.0)
    current_step = Column(String(100), nullable=False, default="Waiting...")
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)

    # Relationships
    user = relationship("User", back_populates="videos", foreign_keys=[user_id])
    course = relationship("Course", back_populates="videos")
    transcripts = relationship("Transcript", back_populates="video", cascade="all, delete-orphan")
    quiz_attempt_videos = relationship("QuizAttemptVideo", back_populates="video", cascade="all, delete-orphan")

File: backend/Models/Video.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VideoIntelligencePlatform.Backend.Models;

[Table("videos")]
public class Video
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("user_id")]
    public int? UserId { get; set; }

    [Column("filename")]
    [MaxLength(255)]
    public string Filename { get; set; } = string.Empty;

    [Column("original_filename")]
    [MaxLength(255)]
    public string OriginalFilename { get; set; } = string.Empty;

    [Column("file_path")]
    [MaxLength(500)]
    public string FilePath { get; set; } = string.Empty;

    [Column("file_size")]
    public long FileSize { get; set; }

    [Column("status")]
    [MaxLength(50)]
    public string Status { get; set; } = "uploaded";

    [Column("progress")]
    public double Progress { get; set; } = 0.0;

    [Column("current_step")]
    [MaxLength(100)]
    public string CurrentStep { get; set; } = "Waiting...";

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("course_id")]
    public int? CourseId { get; set; }

    [Column("order_index")]
    public int OrderIndex { get; set; } = 1;

    [Column("title")]
    [MaxLength(255)]
    public string? Title { get; set; }

    [ForeignKey("UserId")]
    public User? User { get; set; }

    [ForeignKey("CourseId")]
    public Course? Course { get; set; }

    public List<Transcript> Transcripts { get; set; } = new();
    public List<QuizAttemptVideo> QuizAttemptVideos { get; set; } = new();
}
